### PR TITLE
Fix PFC watchdog not getting lossless TC

### PR DESCRIPTION
**Pull request by wendani**
**What I did**

Fix PFC watchdog not getting lossless TCs. This will cause the PFC watchdog not getting started properly if enabled by default. A patch that hard-codes the TCs was used. With this fix, the patch can be removed.
More importantly, it will pave the road to later turn at run time on/off PFC per priority, which may need pfc watchdog to be on/off adjusted accordingly.

This pr is an improvement over https://github.com/Azure/sonic-swss/pull/795

Two techniques are used:
1 ) Allow PFC watchdog startWdOnPort to retry when port lossless TC is empty
2 ) Specify port qos map to be applied last in QosOrch::doTask(), as it must wait for all qos mapping profiles to be ready before rolling out to a port

Log error and return if no lossless TC is found when registering pfcwd stats on a port

**Why I did it**

**How I verified it**

Tested on brcm dut with both cold- and warm-reboot

Also tested with pfcwd warm-reboot test https://github.com/Azure/sonic-mgmt/pull/834
**Details if related**

With technique 1), we can see retry logic in both cold- and warm-reboot because qos map is not ready to apply to ports due to missing qos map profile. And PFC watchdog relies on the pfc_mask info that is written by the qos map installation on a port.

3558:May  3 21:32:49.098087 str--acs-1 ERR swss#orchagent: :- registerInWdDb: No lossless TC found on port Ethernet0
3559-May  3 21:32:49.098087 str--acs-1 ERR swss#orchagent: :- createEntry: Failed to start PFC Watchdog on port Ethernet0
3560-May  3 21:32:49.098087 str--acs-1 ERR swss#orchagent: :- doTask: Failed to process PFC watchdog SET task, retry it
3561:May  3 21:32:49.098087 str--acs-1 ERR swss#orchagent: :- registerInWdDb: No lossless TC found on port Ethernet100
3562-May  3 21:32:49.098087 str--acs-1 ERR swss#orchagent: :- createEntry: Failed to start PFC Watchdog on port Ethernet100
3563-May  3 21:32:49.098087 str--acs-1 ERR swss#orchagent: :- doTask: Failed to process PFC watchdog SET task, retry it
3564:May  3 21:32:49.098163 str--acs-1 ERR swss#orchagent: :- registerInWdDb: No lossless TC found on port Ethernet104
3565-May  3 21:32:49.098163 str--acs-1 ERR swss#orchagent: :- createEntry: Failed to start PFC Watchdog on port Ethernet104
3566-May  3 21:32:49.098163 str--acs-1 ERR swss#orchagent: :- doTask: Failed to process PFC watchdog SET task, retry it

With technique 2), we specify the config execution order in QosOrch to defer port qos mapping to be the last to process. We see no startWdOnPort retry in PFC watchdog in either cold- and warm-reboot. In the warm-reboot case, PFC watchdog port start can finish in iteration 0 of doTask(), while it can only finish in iteration 1 of doTask() without 2).

May  4 00:21:10.449033 str--acs-1 NOTICE swss#orchagent: :- warmRestoreAndSyncUp: OrchDaemon::warmRestoreAndSyncUp: doTask iteration #0

May  4 00:21:11.820573 str--acs-1 NOTICE swss#orchagent: :- processWorkItem: Created [DSCP_TO_TC_MAP:AZURE]
May  4 00:21:11.821866 str--acs-1 NOTICE swss#orchagent: :- processWorkItem: Created [MAP_PFC_PRIORITY_TO_QUEUE:AZURE]
May  4 00:21:11.825710 str--acs-1 NOTICE swss#orchagent: :- handleSchedulerTable: Created [SCHEDULER:scheduler.0]
May  4 00:21:11.825913 str--acs-1 NOTICE swss#orchagent: :- handleSchedulerTable: Created [SCHEDULER:scheduler.1]
May  4 00:21:11.827262 str--acs-1 NOTICE swss#orchagent: :- processWorkItem: Created [TC_TO_PRIORITY_GROUP_MAP:AZURE]
May  4 00:21:11.828551 str--acs-1 NOTICE swss#orchagent: :- processWorkItem: Created [TC_TO_QUEUE_MAP:AZURE]
May  4 00:21:11.828966 str--acs-1 NOTICE swss#orchagent: :- addQosItem: Called create_wred() to create wred profile: oid:0x130000000009cd
May  4 00:21:11.828966 str--acs-1 NOTICE swss#orchagent: :- processWorkItem: Created [WRED_PROFILE:AZURE_LOSSLESS]
May  4 00:21:11.853372 str--acs-1 NOTICE swss#orchagent: :- handlePortQosMapTable: Applied QoS maps to ports

May  4 00:21:14.483174 str--acs-1 NOTICE swss#orchagent: :- createEntry: Started PFC Watchdog on port Ethernet8
May  4 00:21:14.483471 str--acs-1 ERR swss#orchagent: :- doTask: Succeeded PFC watchdog SET task
May  4 00:21:14.487775 str--acs-1 NOTICE swss#orchagent: :- createBindAclTableGroup: Called create_acl_table_group() to create egress ACL table group: oid:0xb000
000000a2b

### CODE REVIEW

File: orchagent/pfcwdorch.cpp [hunk 1]
```diff
@@ -480,6 +502,11 @@ void PfcWdSwOrch<DropHandler, ForwardHandler>::registerInWdDb(const Port& port,
 
         losslessTc.insert(i);
     }
```
Comment (lguohan) [inline orchagent/pfcwdorch.cpp]: what if you get one lossless tc here, and later another lossless tc is added to the port, are you going to retry? I can see you only return false when there is no lossless tc.

File: orchagent/pfcwdorch.cpp [hunk 2]
```diff
@@ -480,6 +502,11 @@ void PfcWdSwOrch<DropHandler, ForwardHandler>::registerInWdDb(const Port& port,
...
+    if (losslessTc.empty())
+    {
+        SWSS_LOG_ERROR("No lossless TC found on port %s", port.m_alias.c_str());
+        return false;
```
Comment (lguohan) [inline orchagent/pfcwdorch.cpp]: there could be a case where no lossless tc is configured, and then pfc wd should not be started in this case. I am not sure you should return ERROR here.

Comment (wendani) [reply orchagent/pfcwdorch.cpp]: If no lossless tc is configured on a port, what is the point of starting pfc watchdog? It is a misconfig in the first place.

File: orchagent/pfcwdorch.cpp (hunk 1 above)
Comment (wendani) [reply orchagent/pfcwdorch.cpp]: This is no processing difference w/ and w/o the PR change regarding the case of another lossless tc being added later to the port.

File: orchagent/pfcwdorch.cpp (hunk 2 above)
Comment (lguohan) [reply orchagent/pfcwdorch.cpp]: not exactly, people might enable pfc watchdog and would expect it is not active until lossless queue is configured on the port.

Comment (wendani) [reply orchagent/pfcwdorch.cpp]: What do you mean by enable pfc watchdog?

**Pull request merged**